tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -185,16 +185,21 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
           classEmitter.genInstanceTests(linkedClass),
           classEmitter.genArrayInstanceTests(linkedClass)
       )(linkedClass.pos)))
+      if(classEmitter.needsSubtypeArray(className)){
+        addTree(classEmitter.genSubtypeArray(linkedClass))
+      }
```

Comment:
This change must be accompanied by the equivalent change in [`ScalaJSClassEmitter.genClassDef`](https://github.com/scala-js/scala-js/blob/v0.6.9/tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala#L96).